pilot/pkg/model/sidecar.go

```diff
@@ -41,7 +41,11 @@ var (
 	}
 
 	sidecarScopeNamespaceConfigTypes = map[resource.GroupVersionKind]struct{}{
-		gvk.Sidecar: {},
+		gvk.Sidecar:               {},
+		gvk.EnvoyFilter:           {},
+		gvk.AuthorizationPolicy:   {},
```

Comment:
I didn't follow the previous PR but remember that EnvoyFilter is additive.
envoyFilter for Workload is (root namespace envoy filters, namespace level envoy filters, plus namespace level workload specific envoy filters (i.e. with matching labels)).

I am not sure if the request authentication is additive or not. You can trace the code to determine the behavior

Reply:
Yeah, that's pretty much the behavior except we'll trigger a push for workload specific filters even if they don't match.

Reply:
Yeah we can technically get even finer grained with the workload selector but I don't think its worth the effort right now - with this change we will support full namespace isolation which is 99% of what we need anyways, and the last 1% will be pretty complex